spring-integration-core/src/main/java/org/springframework/integration/gateway/GatewayProxyFactoryBean.java

```diff
@@ -424,7 +416,7 @@ protected void onInit() {
 
 	@Override
 	public Class<?> getObjectType() {
-		return (this.serviceInterface != null ? this.serviceInterface : null);
+		return this.serviceInterface;
```

Comment:
Why? NPE if `afterPropertiesSet()` is not called.

Reply:
Well, I think we can just remove a setter initialization since there is no any dependency injection overhead with just a `Class<?>` ctor.
Plus that `RequestReplyExchanger.class` could be used from a default ctor. And we don't need that `determineServiceInterface()` logic in the `afterPropertiesSet()`.

I'm OK deferring this to `5.3` and I really regret that it didn't make it into `5.2`, but on the other hand this is fully an internal change and brings some benefit for end-user code.
See that new `verify Mono gateway` test. It doesn't work without some workaround on the end-user side.
We could treat this as a bug, though, because a downstream `Mono` from end-user code is not processed properly in the current version.
But again: I'm OK to move this to `5.3` if you wish.

Reply:
I am ok with merging it, as long as you are confident there are no breaking changes; I suppose we have good test coverage on gateways.

I would say go ahead an initialize the field to RRE and we can get rid of that `determineServiceInterface()` call in aPS.